modules/core/src/main/resources/reference.conf

```diff
@@ -1,4 +1,11 @@
 {
+
+  "gcpUserAgent": {
+    "productName": "Snowplow OSS"
+    "productName": ${?GCP_USER_AGENT_PRODUCT_NAME}
```

Comment:
I don't this syntax exactly but does it mean `productName` will be set to GCP_USER_AGENT_PRODUCT_NAME env variable if it is set ?

Reply:
Yes.  I did it this way because otherwise we would need to change `productName` in three different places in the hocon file:

- `spark.gcpUserAgent`
- `input.gcpUserAgent`
- `output.bad.gcpUserAgent`

But by doing it this way we just need to set one environment variable.

In retrospect, I cannot decide if I like this or not.  It reduces the number of places to set the configuration, but adds some obscurity around how the configuration flows through.  What do you think?

Reply:
I think it is okay to leave it like this.